display-brightness-ddcutil@themightydeity.github.com/convenience.js

```diff
@@ -81,3 +81,66 @@ function spawnWithCallback(argv, callback) {
     }
   });
 }
+
+/**
+ * Check if the given keyval is forbidden.
+ *
+ * @param {number} keyval The keyval number.
+ * @returns {boolean} `true` if the keyval is forbidden.
+ */
+ function isKeyvalForbidden(keyval) {
+  const forbiddenKeyvals = [
+      Gdk.KEY_Home,
+      Gdk.KEY_Left,
+      Gdk.KEY_Up,
+      Gdk.KEY_Right,
+      Gdk.KEY_Down,
+      Gdk.KEY_Page_Up,
+      Gdk.KEY_Page_Down,
+      Gdk.KEY_End,
+      Gdk.KEY_Tab,
+      Gdk.KEY_KP_Enter,
+      Gdk.KEY_Return,
+      Gdk.KEY_Mode_switch,
+  ];
+  return forbiddenKeyvals.includes(keyval);
+}
+
+/**
+ * Check if the given key combo is a valid binding
+ *
+ * @param {{mask: number, keycode: number, keyval:number}} combo An object
+ * representing the key combo.
+ * @returns {boolean} `true` if the key combo is a valid binding.
+ */
+ function isBindingValid({ mask, keycode, keyval }) {
+  if ((mask === 0 || mask === Gdk.SHIFT_MASK) && keycode !== 0) {
+      if (
+          (keyval >= Gdk.KEY_a && keyval <= Gdk.KEY_z) ||
+          (keyval >= Gdk.KEY_A && keyval <= Gdk.KEY_Z) ||
+          (keyval >= Gdk.KEY_0 && keyval <= Gdk.KEY_9) ||
+          (keyval >= Gdk.KEY_kana_fullstop && keyval <= Gdk.KEY_semivoicedsound) ||
+          (keyval >= Gdk.KEY_Arabic_comma && keyval <= Gdk.KEY_Arabic_sukun) ||
+          (keyval >= Gdk.KEY_Serbian_dje && keyval <= Gdk.KEY_Cyrillic_HARDSIGN) ||
+          (keyval >= Gdk.KEY_Greek_ALPHAaccent && keyval <= Gdk.KEY_Greek_omega) ||
+          (keyval >= Gdk.KEY_hebrew_doublelowline && keyval <= Gdk.KEY_hebrew_taf) ||
+          (keyval >= Gdk.KEY_Thai_kokai && keyval <= Gdk.KEY_Thai_lekkao) ||
+          (keyval >= Gdk.KEY_Hangul_Kiyeog && keyval <= Gdk.KEY_Hangul_J_YeorinHieuh) ||
+          (keyval === Gdk.KEY_space && mask === 0) ||
+          isKeyvalForbidden(keyval)
+      )
```

Comment:
We probably need to add `Gdk.KEY_AudioLowerVolume` and `Gdk.KEY_AudioRaiseVolume` to use them, since they are not covered by any of these key ranges.

Reply:
I tried to check that already, this is what I get:
`Mask 0,  Key code 37, Key val 65507`

Which is same as if I just press `Ctrl`.  I do not get any extra information about if Audio Lower or Audio Raise button are pressed along with `Ctrl`. This could of course be some other bug or the limitation of the key event listener in GJS.

Reply:
I will try to check GNOME Control Center source code then and see how it is handled there.

Reply:
We need to inhibit system shortcuts with [inhibit_system_shortcuts](https://docs.gtk.org/gdk4/method.Toplevel.inhibit_system_shortcuts.html) when the dialog is open and restore afterwards with [restore_system_shortcuts](https://docs.gtk.org/gdk4/method.Toplevel.restore_system_shortcuts.html). See [GNOME Control Center implementation](https://gitlab.gnome.org/GNOME/gnome-control-center/-/blob/main/panels/keyboard/cc-keyboard-shortcut-editor.c#L206).

Reply:
Hmm.. don't know how extension review will be with that.  May be an easy fix is to have a text input under the button where you can manully set the shortcuts like before. 

I will check this, just writing to you on what I think.